docs/_snippets/architecture/blocs_domain.dart.md

```diff
@@ -0,0 +1,19 @@
+```dart
+class AppIdeaRankingBloc
+    extends Bloc<AppIdeaRankingEvent, AppIdeaRankingState> {
+  AppIdeaRankingBloc({required AppIdeasRepository appIdeasRepo})
+      : _appIdeasRepo = appIdeasRepo,
```

Comment:
minor but we could remove the `_appIdeasRepo` since it's technically not needed. we could just access the `appIdeasRepo` instance within the event handler.

Reply:
That crossed my mind, but I wasn't sure if the casual use of closure capturing would confuse the reader when they tried to offload the handler to a function, as is sometimes done: `On<MyEvent>(_myFunction)`. It is more concise to omit the repo field variable, though, so I probably should have done that.

Reply:
No problem! I think it's fine as is because it's closer to a real-world implementation. One improvement we could make is to extract the event handler to a private helper but it's not a big deal imo.